clang/lib/Driver/ToolChain.cpp

```diff
@@ -258,6 +259,13 @@ static void getARMMultilibFlags(const Driver &D,
   }
 }
 
+static void getRISCVMultilibFlags(const Driver &D, const llvm::Triple &Triple,
+                                  const llvm::opt::ArgList &Args,
+                                  Multilib::flags_list &Result) {
+  Result.push_back("-march=" + riscv::getRISCVArch(Args, Triple));
```

Comment:
I think that the arch string used here should be normalised using RISCVISAInfo - so that matching is much easier. I think this might throw up some issues when clang changes the version of an extension it implements, but likely you have to check your library build/configuration when that happens anyway, i think.

Reply:
the discourse thread also contains suggestions about splitting up each extension into a different flag - I'm not opposed to an approach like that either, and it might make creating a multilib config which matches the existing behaviour a lot easier.

Reply:
I've made it so that `-march` is now canonicalized. Interestingly, it's already possible to split each extension into its own flag, which I will imminently describe in the discourse thread.